src/sorts/templates/BogoSorting.java

```diff
@@ -28,63 +31,208 @@ of this software and associated documentation files (the "Software"), to deal
  *
  */
 
+/**
+ * Abstract class containing methods commonly used in bogosorts.
+ * <p>
+ * Bogosorts are a subset of impractical sorts
+ * that rely on guess-and-check to sort the array.
+ * This is typically done through randomness.
+ */
 public abstract class BogoSorting extends Sort {
+    /**
+     * The length of each delay, in milliseconds.
+     * <p>
+     * This is set to a small, nonzero number by default
+     * to not affect sorting speed
+     * while still allowing for slowing down the sort.
+     */
+    protected double delay = 1e-9;
+
+    /**
+     * Constructs a new instance of this sort.
+     * @param arrayVisualizer the Array Visualizer instance
+     */
     protected BogoSorting(ArrayVisualizer arrayVisualizer) {
         super(arrayVisualizer);
     }
-    
-    private static int randomPosition(int length, int offset) {
-        return (int) (Math.random() * (length - offset));
+
+    /**
+     * Gets the length of each delay, in milliseconds.
+     * @return the length of delay
+     */
+    public double setDelay() {
```

Comment:
I believe this should be called `getDelay`, not `setDelay`.

Reply:
pure gamer skill